### PR TITLE
Fix unhealtlhy usage of split in language menu

### LINKED ISSUE #5240: Container Overflow when clicking on Options
Hi,
After compiling with the compile.sh Script the game ran and I was able to start a single-player game.
But whenever I click on Options from the main GUI the game stops and this error "==31695==ERROR: AddressSanitizer: container-overflow on address 0x60400027a590 at pc 0x0001050ab553 bp 0x7ff7bee2de90 sp 0x7ff7bee2de88" apears on my terminal.

1. build the game with ./compile.sh on macOS Monterey version 12.2
2. Click on options

**Version:**
 - OS: macOS Monterey version 12.2
 -  Intel processor 
 - This is Widelands Version 1.1~git25630[e9f047d@master]

The whole terminal output is in the log.txt as attachment.

[log.txt](https://github.com/widelands/widelands/files/8048212/log.txt)

Screenshot 2022-02-11 at 12 16 32

### PR DESCRIPTION
**Pull request by klaus-halfmann**
Fixes #5240

### CODE REVIEW

File: src/ui_fsmenu/options.cc [hunk 1]
```diff
@@ -623,12 +623,13 @@ void Options::update_language_stats() {
...
-				if (language_entries_.count(parts[0]) == 1) {
+				std::vector<std::string> parts2;
+				split(parts2, parts[0], {'@'});
+				if (language_entries_.count(parts2[0]) == 1) {
 					locale = parts[0];
 				} else {
-					split(parts, parts[0], {'_'});
-					if (language_entries_.count(parts[0]) == 1) {
+					split(parts, parts2[0], {'_'});
+					if (language_entries_.count(parts2[0]) == 1) {
```
Comment (Noordfrees) [inline src/ui_fsmenu/options.cc]: Shouldn't this be `parts[0]`?

**Comment by klaus-halfmann** [comment]
I cannot set Ahmad-fadl as reviewer

File: src/ui_fsmenu/options.cc (hunk 1 above)
Comment (klaus-halfmann) [reply src/ui_fsmenu/options.cc]: Yes you are correct, I cannot get a ref for i18n::get_locale() and its possible values.

**Review by Ahmad-fadl — APPROVED** [review]
This change fixed the [Isuue](https://github.com/widelands/widelands/issues/5240).
Thanks @klaus-halfmann

**Pull request merged**